### PR TITLE
Add new ECwISC30to60E3r2 ocean and sea-ice mesh

### PR DESCRIPTION
**Pull request by xylar**
Long name: ECwISC30to60L64E3SMv3r2

This mesh is a new eddy closure (EC) mesh with ice-shelf cavities (wISC) that has:
* 30 km resolution at the equator
* 60 km resolution at mid latitudes
* 35 km resolution near the poles

This mesh is a candidate for the E3SM v3 (E3) low res mesh.  It is revision 2 (r2) of the mesh.  (Revision 1 had incorrect bathymetry and coastline around Greenland.)

The mesh was created using [compass](https://github.com/MPAS-Dev/compass), specifically this tag: https://github.com/MPAS-Dev/compass/releases/tag/mesh_ECwISC30to60E3r2

A G-case with data iceberg melt fluxes (DIB) and data ice-shelf melt fluxes (DISMF) has been run for 60 years, with MPAS-Analysis output available here:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.jwolfe/20230901.G-test.ECwISC30to60E3r2.anvil_yrs1-60/html/

The mesh and the G-case results have been reviewed and approved here:
https://acme-climate.atlassian.net/wiki/spaces/OO/pages/3904307201/Review+ECwISC30to60E3r2

A B-case using a CRYO1850 compset with data iceberg melt fluxes (DIB) and data ice-shelf melt fluxes (DISMF) has been run for 20 years, with MPAS-Analysis output available here:
[MPAS-Analysis for: 20230915.test5927.anvil](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.jwolfe/20230915.test5927.anvil/html/)

### CODE REVIEW

**Comment by xylar** [comment]
Some relevant discussion can be found here:

https://github.com/E3SM-Ocean-Discussion/E3SM/pull/63

**Review by proteanplanet — APPROVED** [review]
All critical through-flow areas, shallow bathymetry and coastlines have been checked, along with ice shelf boundaries and cavity geometry.

File: cime_config/config_grids.xml [hunk 1]
```diff
@@ -3290,9 +3362,9 @@
     </gridmap>
 
     <gridmap atm_grid="ne30np4.pg2" ocn_grid="oRRS18to6v3">
-      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_oRRS18to6v3_mono.230806.nc</map>
-      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_oRRS18to6v3_bilin.211101.nc</map>
-      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_oRRS18to6v3_bilin.211101.nc</map>
```
Comment (vanroekel) [inline cime_config/config_grids.xml:3295]: Why are some RRS grid maps changing in the EC30to60 PR?

File: cime_config/config_grids.xml [hunk 2]
```diff
@@ -4610,11 +4721,6 @@
       <map name="ROF2OCN_LIQ_RMAPNAME">cpl/cpl6/map_r05_to_oRRS15to5_nn.160203.nc</map>
     </gridmap>
 
-    <gridmap ocn_grid="oRRS18to6v3" rof_grid="r05">
-      <map name="ROF2OCN_ICE_RMAPNAME">cpl/cpl6/map_r05_to_oRRS18to6v3_smoothed.r50e100.211102.nc</map>
-      <map name="ROF2OCN_LIQ_RMAPNAME">cpl/cpl6/map_r05_to_oRRS18to6v3_smoothed.r50e100.211102.nc</map>
-    </gridmap>
-
```
Comment (vanroekel) [inline cime_config/config_grids.xml:4617]: Same comment, seems unusual to change the RRS maps here

Comment (vanroekel) [reply cime_config/config_grids.xml:4617]: Unless these are both just rebasing issues and will disappear?

Comment (xylar) [reply cime_config/config_grids.xml:4617]: Oh, shoot!  Thanks for pointing that out. I think that was a mistake in a rebase that needs to be fixed.

Comment (xylar) [reply cime_config/config_grids.xml]: @vanroekel, this should be fixed now.  I believe I rebased onto a new `master` and @jonbob already had made changes to the grid files.  Then, when he added his changes after my rebase, it probably inadvertently undid the changes that had happened on `master`.

Thank you for noticing that!

**Comment by xylar** [comment]
@jonbob, please let me know if it is time for me to stage the files produced by compass in `/lcrc/group/e3sm/public_html` yet.  I don't want to jump the gun but I also don't want to make testing unnecessarily difficult.

**Review by mark-petersen — APPROVED** [review]
Reviewed mesh, coastlines, critical passages, and MPAS-Analysis simulation results.

**Comment by jonbob** [comment]
@xylar -- I agree, it's a good time to move the files to the public inputdata repo area

**Comment by xylar** [comment]
@jonbob, everything should be synced over and permissions set to group writable.

**Comment by xylar** [comment]
@jonbob, note that I didn't copy any of the coupling files you made, so I'm hoping you'll copy those.  I was afraid I'd miss something.

**Comment by jonbob** [comment]
@xylar - that's fine, I'll copy the files I generated and maybe try smoke tests on compy or perlmutter to make sure we got everything

**Review by vanroekel — APPROVED** [review]

**Comment by jonbob** [comment]
I made a non-linear mapping file for this mesh, map_ne30pg2_to_ECwISC30to60E3r2_trfvnp2.20230927.nc, and staged it on the inputdata repo. Please note that these new mapping files are not yet included in the config_grids.xml and must be added by hand or in a runscript for testing.

**Comment by jonbob** [comment]
@rljacob and @mt5555 -- I added you as reviewers mostly to get feedback on the updated mapping files. I want to make sure they're compliant with the new procedures and naming convention

**Comment by jonbob** [comment]
passes sanity testing, merged to next

**Comment by xylar** [comment]
@jonbob, wonderful!  Thank you for all the effort you put into this!

**Comment by xylar** [comment]
@jonbob, we need to move files over to `/lcrc/group/e3sm/public_html/inputdata`.  I will move the Compass files now.  Can you make sure to move the coupling files?

**Comment by xylar** [comment]
Okay, I obviously already did that more than a month ago...

**Comment by jonbob** [comment]
And I thought I moved all the files in this PR to the inputdata repo -- @xylar are you finding some missing?

**Comment by jonbob** [comment]
merged to master

**Comment by xylar** [comment]
> And I thought I moved all the files in this PR to the inputdata repo -- @xylar are you finding some missing?

Nope, I just forgot we did that already.  Sorry for the false alarm.

File: cime_config/config_grids.xml [hunk 3]
```diff
@@ -3189,6 +3257,14 @@
       <map name="OCN2ATM_SMAPNAME">cpl/gridmaps/ECwISC30to60E2r1/map_ECwISC30to60E2r1-nomask_to_ne30pg2_mono.201006.nc</map>
     </gridmap>
 
+    <gridmap atm_grid="ne30np4.pg2" ocn_grid="ECwISC30to60E3r2">
+      <map name="ATM2OCN_FMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_ECwISC30to60E3r2_traave.20231018.nc</map>
+      <map name="ATM2OCN_VMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_ECwISC30to60E3r2_trintbilin.20231018.nc</map>
+      <map name="ATM2OCN_SMAPNAME">cpl/gridmaps/ne30pg2/map_ne30pg2_to_ECwISC30to60E3r2-nomask_trintbilin.20231018.nc</map>
+      <map name="OCN2ATM_FMAPNAME">cpl/gridmaps/ECwISC30to60E3r2/map_ECwISC30to60E3r2_to_ne30pg2_traave.20231018.nc</map>
```
Comment (rljacob) [inline cime_config/config_grids.xml:3264]: just noting we ultimately decided to go with just bilin.

Comment (rljacob) [reply cime_config/config_grids.xml:3264]: I mean the algorithm is "bilin". The name would be "trbilin"

**Comment by rljacob** [comment]
@xylar do these have the bug you recently found in the other meshes?

**Comment by xylar** [comment]
Yes, they do.  We're working on a replacement mesh anyway with smoother bathymetry.  So the bug fix will likely come in what that new mesh.

**Comment by xylar** [comment]
For future reference, the bug is: https://github.com/MPAS-Dev/compass/pull/732

**Comment by rljacob** [comment]
Do maps have to be remade?

**Comment by xylar** [comment]
There will need to be maps for the new mesh. I don't see any point in trying to salvage this mesh revision (r2). It will have to be replaced by a new revision, r4 because r3 is already taken for the version without ice-shelf cavities.

**Comment by xylar** [comment]
I don't want to underplay what a screw up this is on my part.  I should have noticed this long ago and it's not a good situation.

But fortunately a relatively few ocean cells are affected.  The ocean flow will occur as normal there but interactions with the atmosphere will be blocked.  I suspect this will not affect coupled simulations with this mesh in a significant way, though local behavior around these covered cells will be affected.

**Pull request merged**